dashboard/src/actions/authActions.js

```diff
@@ -7,6 +7,25 @@ import Cookies from "js-cookie";
 import { showToast } from "actions/toastActions";
 import { uid } from "../utils/helper";
 
+
+// Create an Authentication Request
+export const authenticationRequest = () => async (dispatch, getState) => {
+    const endpoints = getState().apiEndpoint.endpoints;
+    const oidcServer = endpoints?.authentication?.issuer;
+    const oidcRealm = endpoints?.authentication?.realm;
+    const oidcClient = endpoints?.authentication?.client;
+    const oidcClientSecret = endpoints?.authentication?.secret;
+    let req = oidcServer + '/realms/' + oidcRealm + '/protocol/openid-connect/auth';
+    req += '?client_id=' + oidcClient;
```

Comment:
Why do we want to use the server's client ID here?

Reply:
What do you mean? It is only trying to use whatever the pbench-server returned in the endpoints API under `openid-connect` section for the dasboard to connect to the oidc server.

Reply:
But the dashboard is not the same "client" as the "server".

Reply:
Sure, but dashboard doesnt have to know that, all it cares is some configurational stuff it can use to connect to the OIDC server. Now, if the pbench-server thinks that the dashboard should connect on a specific client then it will return those configurations in the endpoints API under `openid-connect`. However, from the dashboard perspective it doesn't make any difference, I mean that was the whole point of having the `openid-connect` section in the endpoints api.

Reply:
I don't think you can have another client pass itself off as a first.  It needs its own client ID.

Reply:
But the Pbench Server knows about the "server" oidc client and the "dashboard" oidc client. Endpoints API is a Pbench-server API. Therefore it can return whatever the server feels appropriate in terms of which "client" the dashboard would need connect to.

Reply:
My conceptualization of this is that the Pbench Server is providing to the Dashboard a set of client credentials to use to access the Keycloak broker.  It's not that the Dashboard and the Pbench Server are separate clients; it's more like the Dashboard is an extension of the Server as far as dealing with the Keycloak broker goes.

I don't think we can configure a different "client" for every instantiation of the Dashboard.  However, we _could_ maintain two client configurations, one for use by the Dashboard and one for use by the Pbench Server, though; and, we may well want to do so:  there will be certain operations which we won't want to trust to the Dashboard (such as changing group memberships and roles), out of concern that some malicious application could obtain the credentials by acting as a Pbench client and then misuse them.  Instead, we'll expect the Dashboard to make a(n authenticated) request to the Pbench Server which will result in the Pbench Server contacting the Keycloak Broker directly -- in that instance we will probably want the Pbench Server to use a different client configuration -- one which is private to the Pbench Server and not shared with the Dashboard.

However, short of groups and roles, we'll need a way for an instance of the Dashboard to be able to authenticate a user with the Keycloak broker, and, in order to do that, it will need to get a client ID and additional credentials, and those will (presumably) come from the Pbench Server.

Reply:
We could do it either way; if we want to create a separate client in our Keycloak configuration, we could, and pass that through to the dashboard.

I'm pretty sure we *don't* want to custom-configure the dashboard code on deployment to use a unique client, and that means it needs to get whatever it needs from `/api/v1/endpoints` when it connects. Once we get to this point, it's just a matter of deciding what data we want to supply it.